src/OpenFeature/Model/FlagMetadata.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+#nullable enable
+namespace OpenFeature.Model;
+
+public sealed class FlagMetadata
```

Comment:
It would be nice to have a more generic metadata class that we could use for `EventMetadata` as well: https://github.com/open-feature/dotnet-sdk/blob/7eebcdda123f9a432a8462d918b7454a26d3e389/src/OpenFeature/Model/ProviderEvents.cs#L39C43-L39C56

The difficulty is that the obvious generic name `Metadata` is already taken.